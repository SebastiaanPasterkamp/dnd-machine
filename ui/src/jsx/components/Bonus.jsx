import React from 'react';
import LazyComponent from '../components/LazyComponent.jsx';

import _ from 'lodash';

class Bonus extends LazyComponent
{
    render() {
        return <div className="bonus inline">
            {this.props.bonus > 0
                ? '+'
                : null
            }
            {!_.isUndefined(this.props.bonus)
                ? this.props.bonus
                : 'â€”'
            }
        </div>;
    }
}

export default Bonus;