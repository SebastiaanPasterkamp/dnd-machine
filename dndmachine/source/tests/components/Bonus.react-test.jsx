import React from 'react';
import Bonus from '../../jsx/components/Bonus.jsx';
import renderer from 'react-test-renderer';

describe('Component: Bonus', () => {
    it('should show mdash for empty bonus', () => {
        const tree = renderer.create(
            <Bonus bonus={null} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should show + before positive bonus', () => {
        const tree = renderer.create(
            <Bonus bonus={1} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should show - before negative bonus', () => {
        const tree = renderer.create(
            <Bonus bonus={-1} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
