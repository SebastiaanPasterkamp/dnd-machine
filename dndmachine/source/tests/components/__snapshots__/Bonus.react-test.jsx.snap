// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Bonus should show + before positive bonus 1`] = `
<div
  className="bonus inline"
>
  +
  1
</div>
`;

exports[`Component: Bonus should show - before negative bonus 1`] = `
<div
  className="bonus inline"
>
  -1
</div>
`;

exports[`Component: Bonus should show mdash for empty bonus 1`] = `
<div
  className="bonus inline"
/>
`;
